homeassistant/components/switch/rest.py

```diff
@@ -97,10 +112,22 @@ def turn_off(self, **kwargs):
 
     def update(self):
         """Get the latest data from REST API and update the state."""
-        request = requests.get(self._resource, timeout=10)
-        if request.text == self._body_on:
-            self._state = True
-        elif request.text == self._body_off:
-            self._state = False
+        request = requests.get(self._resource, timeout=self._timeout)
+
+        if self._value_template is not None:
+            response = template.render_with_possible_json_value(
+                self._hass, self._value_template, request.text, 'None')
+            response = response.lower()
+            if response == 'true':
```

Comment:
I can explain the current semantics in the page at home-assistant.io and I think it won't be too complicated or confusing. The current choice is mainly to ensure backward compatibility.
Alternatively, as you are suggesting (IIUC), we can force the user to always provide `value_template`, which computes the state of the switch while polling. Then we may want to automatically upgrade existing config files to add `value_template: {{ value.json == body_on }}`.
I like the latter approach, but I don't know if it can be accomplished easily.
Also, perhaps `value_template` should be renamed to `is_on`.